### ISSUE TITLE
`unique()` is not intuitively understandable.

### ISSUE THREAD

**Issue by hyperkai** [pytorch/pytorch#130217]
### 📚 The doc issue

[The doc](https://pytorch.org/docs/stable/generated/torch.unique.html) of `unique()` says that `return_inverse=True` returns the indices for where elements in the original input ended up in the returned unique list as shown below:

> - input ([Tensor](https://pytorch.org/docs/stable/tensors.html#torch.Tensor)) – the input tensor

>- sorted ([bool](https://docs.python.org/3/library/functions.html#bool)) – Whether to sort the unique elements in ascending order before returning as output.

> - return_inverse ([bool](https://docs.python.org/3/library/functions.html#bool)) – Whether to also return the indices for where elements in the original input ended up in the returned unique list.

> - return_counts ([bool](https://docs.python.org/3/library/functions.html#bool)) – Whether to also return the counts for each unique element.

> - dim ([int](https://docs.python.org/3/library/functions.html#int), optional) – the dimension to operate upon. If None, the unique of the flattened input is returned. Otherwise, each of the tensors indexed by the given dimension is treated as one of the elements to apply the unique operation upon. See examples for more details. Default: None

But the indices are not returned while `return_counts=True` properly returns the couns for each unique element as shown below:

```python
import torch

my_tensor = torch.tensor([[[2, 2, 0], [0, 1, 1]],
                          [[1, 3, 0], [0, 0, 2]]])
torch.unique(input=my_tensor, return_inverse=True, return_counts=True)
# (tensor([0, 1, 2, 3]),
# (tensor([0, 1, 2, 3]),
#  tensor([[[2, 2, 0],
#           [0, 1, 1]], 
#          [[1, 3, 0],
#           [0, 0, 2]]]),
#  tensor([5, 3, 3, 1]))
```

### Suggest a potential alternative/fix

```python
torch.__version__

torch.__version__ # 2.3.0+cu121
```
